### PR TITLE
style option to enable background color

### PR DESCRIPTION
**Pull request by WuerfelDev**
This pr lets you set the `style=background` option. It will get the background color from the theme.
Also addresses #365

### Tests
I'm having issues fixing the snapshot test "full". The generated file and the test result don't match. I found that the problem are some whitespaces:
```diff
--- /tmp/bat_full_actual
+++ /tmp/bat_full_expected
@@ -15,7 +15,7 @@
   11 ~ │ [48;2;34;34;34m        [0m[48;2;34;34;34m"[0m[48;2;34;34;34mThe perimeter of the rectangle is [0m[48;2;34;34;34m{}[0m[48;2;34;34;34m pixels.[0m[48;2;34;34;34m"[0m[48;2;34;34;34m,[0m[48;2;34;34;34m                [0m
   12 ~ │ [48;2;34;34;34m        [0m[48;2;34;34;34mperimeter[0m[48;2;34;34;34m([0m[48;2;34;34;34m&[0m[48;2;34;34;34mrect1[0m[48;2;34;34;34m)[0m[48;2;34;34;34m                                              [0m
   13   │ [48;2;34;34;34m    [0m[48;2;34;34;34m)[0m[48;2;34;34;34m;[0m[48;2;34;34;34m                                                                 [0m
-  14 + │ [48;2;34;34;34m    [0m[48;2;34;34;34mprintln![0m[48;2;34;34;34m([0m[48;2;34;34;34mr[0m[48;2;34;34;34m#[0m[48;2;34;34;34m"This line contains invalid utf8:  "�����[0m[48;2;34;34;34m"#[0m[48;2;34;34;34m;[0m[48;2;34;34;34m            [0m
+  14 + │ [48;2;34;34;34m    [0m[48;2;34;34;34mprintln![0m[48;2;34;34;34m([0m[48;2;34;34;34mr[0m[48;2;34;34;34m#[0m[48;2;34;34;34m"This line contains invalid utf8:  "�����[0m[48;2;34;34;34m"#[0m[48;2;34;34;34m;[0m[48;2;34;34;34m  [0m
   15   │ [48;2;34;34;34m}[0m[48;2;34;34;34m                                                                      [0m
   16   │ [48;2;34;34;34m                                                                       [0m
   17   │ [48;2;34;34;34mfn[0m[48;2;34;34;34m [0m[48;2;34;34;34marea[0m[48;2;34;34;34m([0m[48;2;34;34;34mrectangle[0m[48;2;34;34;34m:[0m[48;2;34;34;34m [0m[48;2;34;34;34m&[0m[48;2;34;34;34mRectangle[0m[48;2;34;34;34m)[0m[48;2;34;34;34m [0m[48;2;34;34;34m->[0m[48;2;34;34;34m [0m[48;2;34;34;34mu32[0m[48;2;34;34;34m [0m[48;2;34;34;34m{[0m[48;2;34;34;34m                                [0m

```

### Further improvements (?)

Don't show background colors (also no highlight bg) when `--color=never` is set. Would like to hear an opinion about that.
This would likely fix the bug above because of https://github.com/sharkdp/bat/blob/52ef1bc0d6bead9f450efe28a457acbcb1e94d88/tests/snapshots/generate_snapshots.py#L26

### CODE REVIEW

File: assets/completions/bat.fish.in [hunk 1]
```diff
@@ -85,7 +85,8 @@ function __bat_style_opts
         "grid,lines b/w sidebar/header/content" \
         "numbers,line numbers in sidebar" \
         "rule,separate files" \
-        "snip,separate ranges"
+        "snip,separate ranges" \
+        "backgroundd,theme background color"
```
Comment (sharkdp) [inline assets/completions/bat.fish.in:89]: typo

**Comment by sharkdp** [comment]
> I'm having issues fixing the snapshot test "full". The generated file and the test result don't match. I found that the problem are some whitespaces

I think that is expected, because if you add a background color (as opposed to setting it to `None`), `bat` will have to print whitespace characters to "complete" the line:

https://github.com/sharkdp/bat/blob/a422fea06d7e3c3477760382996b8b0ba9f3442c/src/printer.rs#L551-L563

This is also the main reason why I rejected the initial ticket. Adding background colors means that `bat` will not just print file contents, but additional newlines. This is somewhat acceptable for `--highlight-line` IMO, but I'm still not quite sure about a generic `--style=background` option.

**Comment by WuerfelDev** [comment]
> This is also the main reason why I rejected the initial ticket. Adding background colors means that `bat` will not just print file contents, but additional newlines. This is somewhat acceptable for `--highlight-line` IMO, but I'm still not quite sure about a generic `--style=background` option.

Understood. I didn't see it as a problem since it's not used by default and has to be enabled manually. Other style options (decorations) change the output too - I don't understand how background would be different from that.

Also I found an alternative way of setting a background without needing to print whitespaces here: https://stackoverflow.com/a/62371559

**Comment by sharkdp** [comment]
I'm sorry, but I would like to close this. This is not something typical CLI applications provide (TUI yes, but not CLI). And those additional spaces can lead to all sorts of complications.

Thank you for your contribution though.

**Comment by WuerfelDev** [comment]
My main reason with the PR was to address #365 since I also use a transparent background and find `bat`s output generally hard to read. 
And there was already a background color given by themes so why not use ^^

**Pull request closed without merging**